### ISSUE TITLE
Support limiting symbols to a particular vintage character set

### ISSUE THREAD

**Issue by kimslawson** [hpjansson/chafa#110]
It would be a nice addition to chafa to support restricting the characters used to a subset corresponding to a particular computer's character set (for example, PETSCII, ATASCII, Apple II, etc.) While the `vintage` class to symbols and fill is nice, it's not specific enough... If I'm making ATASCII art, I don't want it to include characters that aren't supported by the Atari's limited character set.

I'd like to see classes corresponding to the individual computer/terminal types, so as to limit the output to that particular character set. (for example `--symbols=ATASCII`, `--fill=PETSCII`)

Specifically, characters from each supported character set class would need to be mapped to the appropriate unicode points, and those sets assigned to individual symbol classes. [Unscii is a font](http://viznut.fi/unscii/) that might help with this endeavour, as well as displaying the output in a terminal, as it faithfully tries to recreate the pixelated character sets of many retro platforms.

([Related closed issue here](https://github.com/hpjansson/chafa/issues/19))

**Comment by hpjansson** [comment]
Yes! I actually want to implement a `--preset` (`-P`) option for this with built-in presets for vintage platforms. I've made rudimentary starts for [C64](https://github.com/hpjansson/chafa/blob/master/tools/chafa/c64-platform.c) and [IBM/VGA/cp437](https://github.com/hpjansson/chafa/blob/master/tools/chafa/ibm-platform.c). A little more groundwork needs to be laid so it can also emulate the various color palettes.

**Comment by kimslawson** [comment]
Awesome! 

<img width="650" alt="Screen Shot 2022-09-16 at 11 43 32" src="https://user-images.githubusercontent.com/249884/190678348-ea55bc4f-7c97-4807-b501-c8b9dfaeed0a.png">

This is what I get when I test using unscii-8 right now, but for example ATASCII has no support for many of those glyphs.

**Comment by hpjansson** [comment]
Nice Workbench 1.3 vibes :-)

By the way, you can specify individual glyphs by passing their hex codes to `--symbols`, e.g. `--symbols 20,30..39`, but it's obviously pretty tedious.

**Comment by kimslawson** [comment]
> Nice Workbench 1.3 vibes :-)

Ha. I was going for Atari mode 0 but I got the luminance wrong.

**Comment by kimslawson** [comment]
> This is what I get when I test using unscii-8 right now, but for example ATASCII has no support for many of those glyphs.

<img width="730" alt="Screen Shot 2022-09-16 at 12 29 13" src="https://user-images.githubusercontent.com/249884/190686515-971a4416-5095-46eb-985e-feedb74de822.png">


of course I haven't had enough coffee, because I got that command wrong. THIS is what it should look like with it restricted to `legacy` (I had it mixed up with "vintage" which is a nonexistent class)

Relevant still is that some of those are not supported by certain systems. So the point still stands :)

**Comment by hpjansson** [comment]
Indeed.

I'd love to have some help with the research. For instance, I'd accept a PR with an ATASCII table similar to the ones I did for C64/cp437. Or maybe there needs to be more than one table -- I'm not sure what the differences are between the Atari 400 and later models.

**Comment by kimslawson** [comment]
> similar to the ones I did for C64/cp437

I had a look around the code and docs, but didn't see this. Can you point me in the right direction?

**Comment by hpjansson** [comment]
Ah sorry, I linked to the source files in https://github.com/hpjansson/chafa/issues/110#issuecomment-1249519684. They're not used for anything yet, but it's useful to keep adding those so I can integrate them later.

**Comment by clort81** [comment]
> > This is what I get when I test using unscii-8 right now, but for example ATASCII has no support for many of those glyphs.
> 
> <img alt="Screen Shot 2022-09-16 at 12 29 13" width="730" src="https://user-images.githubusercontent.com/249884/190686515-971a4416-5095-46eb-985e-feedb74de822.png">
> 
> of course I haven't had enough coffee, because I got that command wrong. THIS is what it should look like with it restricted to `legacy` (I had it mixed up with "vintage" which is a nonexistent class)
> 
> Relevant still is that some of those are not supported by certain systems. So the point still stands :)

Amazingly, Viznut failed to add the eight glyphs that would be needed to emulate PETSCII triangles in half-width (common aspect ratio) monospace fonts.   For that I made 8 glyphs with double-wide triangles, so that Blapinus has glyphs with 45 degree angle triangles.

**Comment by clort81** [comment]
A HUGE THANK YOU to hpjansson for now restricting chosen glyphs to those contained in the  --glyph-file !
```
/usr/local/bin/chafa --color-space=din99d --color-extractor=median -p on --symbols all --glyph-file /usr/share/fonts/X11/misc/blapetscii-12.pcf.gz -w 8 -s 80 redcrab-ihasfloppy.png```
Now i can limit the glyphs used from my real font (blapinus).

Would anyone  consider a pull request that reads-in a list of glyph codepoints from a text file, and ONLY uses those?

Without the --glyph-file it is impossible for me to set the range that I want explicitly on the command line, using, for example
```
chafa -symbols none+u2580+u2584+E000-E07F -s $width $myfile
```
there ought to be a way for user to specify the **exact codepoint ranges** to use upon invocation and not get overriden by inscrutable internal program range-selection logic.

**Comment by Marrin** [comment]
@hpjansson Why the screen codes for the C64 and not PETSCII?  How would colors be encoded then and other conrol codes for clearing the screen or sending the cursor to the upper left?
